### PR TITLE
Performance improvement

### LINKED ISSUE #74: Add synthetic traffic detection for Keynote
We currently support such user agent strings:
_Mozilla/4.0 (compatible; MSIE 6.0; Windows NT 5.0;_ **KTXN**_)_ which is used for the real browser Transaction Perspective monitors.

Also we need to consider the below keynote traffic as well.
•"KHTE" (for Application Perspective monitors), 
•"Keynote" for Test Perspective load testing agents.

### PR DESCRIPTION
**Pull request by abaranch**
Replace regex on "contains" which works 50% faster.
Remove duplicate filters (e.g. "ABC bot" and "DEF bot" replaced on "bot")
Also resolve #74

### CODE REVIEW

**Comment by Dmitry-Matveev** [comment]
:shipit:

File: Src/Web/Web.Shared.Net.Tests/SyntheticUserAgentTelemetryInitializerTest.cs [hunk 1]
```diff
@@ -5,29 +5,12 @@
...
-            SourceName = "Spider"
-        };
+        private string botRegex = "search|spider|crawl|Bot|Monitor|BrowserMob|BingPreview|PagePeeker|WebThumb|URL2PNG|ZooShot|GomezA|Google SketchUp|Read Later|KTXN|KHTE|Keynote|Pingdom|AlwaysOn|zao|borg|oegp|silk|Xenu|zeal|NING|htdig|lycos|slurp|teoma|voila|yahoo|Sogou|CiBra|Nutch|Java|JNLP|Daumoa|Genieo|ichiro|larbin|pompos|Scrapy|snappy|speedy|vortex|favicon|indexer|Riddler|scooter|scraper|scrubby|WhatWeb|WinHTTP|voyager|archiver|Icarus6j|mogimogi|Netvibes|altavista|charlotte|findlinks|Retreiver|TLSProber|WordPress|wsr-agent|http client|Python-urllib|AppEngine-Google|semanticdiscovery|facebookexternalhit|web/snippet|Google-HTTP-Java-Client";
```
Comment (dnduffy) [inline Src/Web/Web.Shared.Net.Tests/SyntheticUserAgentTelemetryInitializerTest.cs]: nitpick: it's not a regular expression anymore so maybe a name change would be clearer.

File: Src/Web/Web.Shared.Net.Tests/SyntheticUserAgentTelemetryInitializerTest.cs [hunk 2]
```diff
@@ -39,210 +22,166 @@ public void SyntheticSourceIsNotSetIfUserProvidedValue()
...
-                Pattern = "(YottaaMonitor|BrowserMob|HttpMonitor|YandexBot|BingPreview|PagePeeker|ThumbShotsBot|WebThumb|URL2PNG|ZooShot|GomezA|Catchpoint bot|Willow Internet Crawler|Google SketchUp|Read%20Later)",
-                SourceName = "Robot"
-            });
+            source.Filters = this.botRegex;
 
             source.Initialize(metricTelemetry);
 
             Assert.IsNull(metricTelemetry.Context.Operation.SyntheticSource);
-        }
+        }
```
Comment (dnduffy) [inline Src/Web/Web.Shared.Net.Tests/SyntheticUserAgentTelemetryInitializerTest.cs]: nitpick: extra spaces.

**Comment by dnduffy** [comment]
Good changes. :) :shipit:

**Comment by AlexBulankou** [comment]
:+1:

**Pull request merged**